### PR TITLE
AVRO-3458: Added tests for GenericRecord

### PR DESCRIPTION
**Pull request by KyleSchoonover**
### Jira

- [ ] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO/) issues and references them in the PR title. For example, "AVRO-1234: My Avro PR"
  - https://issues.apache.org/jira/browse/AVRO-3458

### Tests

- [ ] My PR does not need testing for this extremely good reason: It is just adding tests

### Commits

- [ ] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

**Comment by zcsizmadia** [comment]
1. GenericTests.cs already has many GenerricRecord related tests. Why are these tests in a seperate GenericRecordTests.cs file?
2. Could these tests utilize the mkRecord and other helper code implemenetd in GenericTests.cs?
3. schema definition strings are preferred to be defined as const/readonly/static fileds or use test case to specify them

**Comment by KyleSchoonover** [comment]
> 1. GenericTests.cs already has many GenerricRecord related tests. Why are these tests in a seperate GenericRecordTests.cs file?
> 2. Could these tests utilize the mkRecord and other helper code implemenetd in GenericTests.cs?
> 3. schema definition strings are preferred to be defined as const/readonly/static fileds or use test case to specify them

1. These are unit tests specific to just GenericRecord.
2.  It could, but the tests don't require something that complex
3. Moved the base schema to a const.

**Comment by KyleSchoonover** [comment]
> 1. GenericTests.cs already has many GenerricRecord related tests. Why are these tests in a seperate GenericRecordTests.cs file?
> 2. Could these tests utilize the mkRecord and other helper code implemenetd in GenericTests.cs?
> 3. schema definition strings are preferred to be defined as const/readonly/static fileds or use test case to specify them

1. These are unit tests specific to just GenericRecord.
2.  It could, but the tests don't require something that complex
3. Moved the base schema to a const.

**Review by zcsizmadia — CHANGES_REQUESTED** [review]

File: lang/csharp/src/apache/test/Generic/GenericRecordTests.cs
Comment (KyleSchoonover) [reply lang/csharp/src/apache/test/Generic/GenericRecordTests.cs]: Changed to discard

**Comment by zcsizmadia** [comment]
I usually wait for the checks to go green before approval. Can you fix the failed checks plz?

**Review by zcsizmadia — APPROVED** [review]

**Comment by martin-g** [comment]
Thank you, @KyleSchoonover !

**Pull request merged**
